src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -57,7 +57,8 @@ public class AutoRepairConfig implements Serializable
     public enum RepairType implements Serializable
     {
         full,
-        incremental;
+        incremental,
+        preview_repaired;
```

Comment:
caught an exception that is a side effect of the recent changes to make things configured at the table level:

```
ERROR [AutoRepair-Repair-PREVIEW_REPAIRED:1] 2024-11-25 21:02:41,568 AutoRepair.java:342 - Exception in autorepair:
java.lang.NullPointerException: null
        at org.apache.cassandra.repair.autorepair.AutoRepair.retrieveTablesToBeRepaired(AutoRepair.java:380)
        at org.apache.cassandra.repair.autorepair.AutoRepair.repair(AutoRepair.java:216)
        at org.apache.cassandra.repair.autorepair.AutoRepair.lambda$setup$0(AutoRepair.java:131)
        at org.apache.cassandra.concurrent.ExecutionFailure$1.run(ExecutionFailure.java:133)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

I think we need to plumb through something similar for `preview_repaired` for the table options, see `TableParams` for a good starting point:  https://github.com/kzalys/cassandra/blob/auto_repair_v2_on_4_1/src/java/org/apache/cassandra/schema/TableParams.java#L63-L64